### ISSUE TITLE
Automated graph and collection creation

### ISSUE THREAD

**Issue by slomo** [facette/facette#227]
Hi,

I am setting up my cluster (and also facette) using saltstack, which is configuration management system comparable to puppet or chef. Which means I know exactly in advance which graphs I want to create. Is there better way than the HTTP-API to create graphs (Maybe like adding it to the config file or a command line utility).

**Comment by vbatoufflet** [comment]
Hi,

Today, there is no other way to create graphs other than using the HTTP API. I doubt we implement a way to do it via configurations, but using a command-line utility (through `facettectl` by example) could be interesting.

In the meantime, using Salt myself, I was planning one day to create a state for Facette. But I didn't had the time to do it yet.

Maybe the [python-facett](https://github.com/OpenTouch/python-facette) could be useful for that. But I didn't test it for now :smile:

**Comment by psytra** [comment]
I've been wondering if it is necessary to have all graphs statically created.
Woudln't it be nice if we could simply have templates and add template variables to the url string like
/browse/collections/54ecb213-b14e-4e3d-65cb-2a251afb659f?hostname=myhostname 
of course, you'll still need to decide which templates are relevant for which hosts...

Right now, using facette as a drop-in replacement for, let's say cgp requires API scripting which might be cumbersome.

**Comment by trehn** [comment]
Big +1 on this. I'm already doing it (see #233 for details), but of course thats all kinds of unsupported ;)

**Comment by kiranos** [comment]
we use CGP today as its pretty much handles itself once you set it up and have all "clients" export graphs to CGP, having this feature in facette is a must for me, as we create servers very often. Its to tedious to do things manually over and over :)

**Comment by slomo** [comment]
I am using CGP now instead. At it works badly ... but my company won't prio this high enough to let me work on something for facette. Unless some else wants to step forward here I would say the ticket  should be closed.

**Comment by kiranos** [comment]
I dont agree on closing this, A big reason why I'm not using facette today (plus devs say its not suited for production use) is that it takes along time to set up graphs for each host, having an "wizard" or template function if you have many servers that use the same collectd plugins would make facette much easier to use. And I think would create more active users. But as its not recommended for production use yet, I dont think its a priority, but it should be in none the less.

**Comment by blainsmith** [comment]
I know this is an old issue, but I agree with @kiranos. I was hoping to use this for IoT device health and I have over 5,000 devices I can potentially collect from `collectd` and creating things by hand or having a script create them when a new host comes online is error prone. I would have a collection for every host and the same 5 graphs for every one too.
